packages/remote-replay-launcher/src/index.ts

```diff
@@ -155,6 +151,33 @@ export const executeRemoteTestRun = async ({
 
   const completedTestRun = await testRunCompleted.promise;
 
+  const tunnelUnlocked = defer<void>();
+  let tunnelCheckInterval: NodeJS.Timeout | undefined = undefined;
+  const checkUnlocked = async () => {
+    const isLocked = await getIsLocked({ client, deploymentId });
+    if (isLocked) {
+      onTunnelLocked?.();
+      return false;
+    }
+    if (tunnelCheckInterval) {
+      clearInterval(tunnelCheckInterval);
+    }
+    if (keepTunnelOpenPromise) {
+      void keepTunnelOpenPromise.then(() => {
+        tunnel.close();
+      });
+    } else {
+      tunnel.close();
+    }
+    tunnelUnlocked.resolve();
+    return true;
+  };
+  const alreadyUnlocked = await checkUnlocked();
+  if (!alreadyUnlocked) {
+    tunnelCheckInterval = setInterval(checkUnlocked, POLL_LOCK_INTERVAL_MS);
+    await tunnelUnlocked.promise;
+  }
+
```

Comment:
The `keepTunnelOpenPromise` logic and `tunnelCheckInterval` clearing can be moved outside of this to improve readability.
```suggestion

    tunnelUnlocked.resolve();
    return true;
  };
  const alreadyUnlocked = await checkUnlocked();
  if (!alreadyUnlocked) {
    tunnelCheckInterval = setInterval(checkUnlocked, POLL_LOCK_INTERVAL_MS);
    await tunnelUnlocked.promise;
  }

  if (tunnelCheckInterval) {
    clearInterval(tunnelCheckInterval);
  }
  if (keepTunnelOpenPromise) {
    await keepTunnelOpenPromise;
  }

  tunnel.close();

```

Reply:
That's much cleaner, thank you.